distributed/http/proxy.py

```diff
@@ -13,10 +13,43 @@ class GlobalProxyHandler(ProxyHandler):
         from a port to any valid endpoint'.
         """
 
+        def is_worker_host(self, proxy_handler, host=None):
+            """
+            Check whether host is a valid worker hostname
+
+            This is necessary to ensure that the Dashboard can actually proxy requests
+            to worker dashboards when running a distributed cluster where each worker
+            may be on a different hostname than the scheduler host
+
+            Parameters
+            ----------
+
+            host : str
+
+            Returns
+            -------
+            bool
+            """
+            if host is None:
+                return False
+            if self.scheduler is None:
+                return False
+            workers = list(self.scheduler.workers.values())
```

Comment:
If all you need is an iterable I think the `list` is redundant:
```suggestion
            workers = self.scheduler.workers.values()
```

Reply:
Or you could inline this with the line below

```python
for w in self.scheduler.workers.values():
```

This is how most of the other code looks so far

Reply:
Actually it looks like you're maybe looking for the following

```python
return any(w.host == host for w in self.scheduler.workers.values())
```